src/commands/ImportUriHandler.ts

```diff
@@ -0,0 +1,41 @@
+import * as querystring from 'querystring';
+import { Disposable, Uri, UriHandler, window } from 'vscode';
+import { ext } from 'vscode-azureappservice/out/src/extensionVariables';
+import { callWithTelemetryAndErrorHandling, IActionContext } from '../../extension.bundle';
+import { localize } from '../localize';
+import { importTrialApp } from './trialApp/importTrialApp';
+
+export class ImportUriHandler implements UriHandler {
+    private disposables: Disposable[] = [];
+
+    constructor() {
+        this.disposables.push(window.registerUriHandler(this));
```

Comment:
I might be wrong, but could you not do this in the `extension.ts` file?

`context.subscriptions.push(new ImportUriHandler());`

`context.subscriptions` is an array of disposals that get cleaned up after the extension is deactivated.  You could remove some of this extra code then.